internal/system/agent/init.go

```diff
@@ -100,6 +100,7 @@ func Init(useRegistry bool) bool {
 	if Configuration == nil {
 		return false
 	}
+	executorClient = &ExecuteApp{}
```

Comment:
- Please confirm your use of `executorClient` (the empty interface) since it, in conjunction with `type ExecuteApp struct`, no longer performs any (useful) functionality...

Reply:
- Other than this (and other references to `executorClient`), the PR LGTM.

Reply:
Note that the `executorClient` is _of type_ `interface {}`, which means that it could hold any type (or also be `nil`). It importantly does perform useful functionality because this `executorClient` variable is used to actually run commands here: https://github.com/edgexfoundry/edgex-go/blob/4d274133c690b91bd282f994267e04970c2eb073/internal/system/agent/services.go#L49 we attempt to type assert this variable into the various operation interfaces. If we never initialize this variable like being done here, then when we get to that snippet of code above, `executorClient` is nil, and as such can't match the interface of say `ServiceStarter`, and as such the operation fails to be executed.

It's not clear to me the code could ever have worked without this initialization of the `executorClient` variable. 

Also note that while I did originally ask that you remove the function initializing this variable, I only meant that the _function_ was useless because it was just assigning a variable, I did not mean the variable was useless. Apologies if that comment on your earlier PR was unclear.

Reply:
The detailed response is very helpful @anonymouse64 👍 - I'll catch up with you offline, so we can get this squared away, thanks.

Reply:
- Was trying to catch up and chat... In summary, please proceed with the initialization of the `executorClient` variable (the way you have it currently should work).